core/src/test/java/com/graphhopper/routing/subnetwork/PrepareRoutingSubnetworksTest.java

```diff
@@ -83,103 +71,111 @@ private static GraphHopperStorage createSubnetworkTestStorage(String flagEncoder
-        BooleanEncodedValue inSubnetwork = em.getBooleanEncodedValue(Subnetwork.key(str));
-        BooleanEncodedValue accessEnc = em.getEncoder(str).getAccessEnc();
-        return e -> !e.get(inSubnetwork) && e.get(accessEnc);
+        assertEquals(0, instance.doWork());
+        assertEquals(IntArrayList.from(), getSubnetworkEdges(g, encoder));
     }
 
     @Test
-    public void testRemoveSubnetworkIfOnlyOneVehicleEdgeBased() {
-        GraphHopperStorage g = createSubnetworkTestStorage("car|turn_costs=true");
-        EncodingManager em = g.getEncodingManager();
-        FlagEncoder encoder = g.getEncodingManager().fetchEdgeEncoders().iterator().next();
+    public void testPrepareSubnetworkWithTurnCosts() {
+        EncodingManager em = createEncodingManager("car|turn_costs=true");
+        FlagEncoder encoder = em.fetchEdgeEncoders().iterator().next();
+        GraphHopperStorage g = createSubnetworkTestStorage(em, encoder);
         PrepareRoutingSubnetworks instance = new PrepareRoutingSubnetworks(g, Collections.singletonList(
                 createJob(em, encoder, new DefaultTurnCostProvider(encoder, g.getTurnCostStorage(), 0))));
-        // this rules out the upper small network
+        // this will make the upper small network a subnetwork
         instance.setMinNetworkSize(4);
-        instance.doWork();
-        assertTrue(GHUtility.getProblems(g).isEmpty());
-        EdgeExplorer explorer = g.createEdgeExplorer(createFilter(em, encoder.toString()));
-        assertEquals(GHUtility.asSet(), GHUtility.getNeighbors(explorer.setBaseNode(4)));
+        assertEquals(3, instance.doWork());
+        assertEquals(IntArrayList.from(7, 8, 9), getSubnetworkEdges(g, encoder));
 
         // this time we lower the threshold and the small network will remain
-        g = createSubnetworkTestStorage("car|turn_costs=true");
-        em = g.getEncodingManager();
-        encoder = g.getEncodingManager().fetchEdgeEncoders().iterator().next();
+        em = createEncodingManager("car|turn_costs=true");
+        encoder = em.fetchEdgeEncoders().iterator().next();
+        g = createSubnetworkTestStorage(em, encoder);
         instance = new PrepareRoutingSubnetworks(g, Collections.singletonList(createJob(em, encoder, NO_TURN_COST_PROVIDER)));
         instance.setMinNetworkSize(3);
-        instance.doWork();
-        explorer = g.createEdgeExplorer(createFilter(em, encoder.toString()));
-        assertEquals(GHUtility.asSet(5, 6), GHUtility.getNeighbors(explorer.setBaseNode(4)));
+        assertEquals(0, instance.doWork());
+        assertEquals(IntArrayList.from(), getSubnetworkEdges(g, encoder));
     }
 
     @Test
-    public void testRemoveNode() {
-        GraphHopperStorage g = createSubnetworkTestStorage("car,bike");
-        EncodingManager em = g.getEncodingManager();
-        FlagEncoder carEncoder = g.getEncodingManager().getEncoder("car");
-        FlagEncoder bikeEncoder = g.getEncodingManager().getEncoder("bike");
```

Comment:
This test tested a method that was not used anyway and did not even use PrepareRoutingSubnetworks to prepare the subnetworks, so it basically tested nothing :) I probably forgot deleting it in a previous refactoring.